### PR TITLE
Use template to display on demand edit form

### PR DESCRIPTION
**Pull request by kiran-mastek**
Story [AB#9131](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/9131)
Task [AB#11902](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/11902)

### CODE REVIEW

File: app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/manifest.json
Comment (bjss-chris) [inline app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/manifest.json:6]: As the manifest is just a JSON object that we use to supply data to the template, templating has no effect here so we may as well just use a literal placeholder, e.g. `"unit description"`.

File: app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/manifest.json
Comment (bjss-chris) [inline app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/manifest.json:83]: Similar to above, I think we'll need to set the selected estimation period in the `contextCreator`. However, we won't be able to do that until we have the previous page in the flow (quantity and estimation period) that @nehajain008 is currently working on. Therefore, I think it will be fine to leave this as is for now as a prompt.

File: app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/general.ui.test.js
Comment (bjss-chris) [inline app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/general.ui.test.js:20]: It would be better to use an example of an on-demand price here; an option might be consultations, for example: 

```javascript
itemUnit: {
    name: 'consultationCore',
    description: 'per consultation – core hours',
}
```

File: app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/general.ui.test.js
Comment (bjss-chris) [inline app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/general.ui.test.js:25]: On-demand prices don't have a time unit so we can delete the `timeUnit` property.

File: app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/general.ui.test.js
Comment (bjss-chris) [inline app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/general.ui.test.js:95]: Practice size is specific to per patient pricing so we should change this to `quantityColumnHeading`.

File: app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/general.ui.test.js
Comment (bjss-chris) [inline app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/general.ui.test.js:267]: Similar to above, `practice list sizes` should be `quantities`.

File: app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/general.ui.test.js
Comment (bjss-chris) [inline app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/general.ui.test.js:283]: Similar to above, `quantity` instead of `practice size`.

File: app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/general.ui.test.js
Comment (bjss-chris) [inline app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/general.ui.test.js:116]: Similar to above, we should rename this `const` to `quantityExpandableSection`.

File: app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/withSavedData.ui.test.js
Comment (bjss-chris) [inline app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/withSavedData.ui.test.js:28]: As with the general UI test, it would be better to use an example of an on-demand price here; an option might be consultations, for example: 

```javascript
itemUnit: {
    name: 'consultationCore',
    description: 'per consultation – core hours',
}
```

File: app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/withSavedData.ui.test.js
Comment (bjss-chris) [inline app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/withSavedData.ui.test.js:33]: As with the general UI test, on-demand prices don't have a time unit so we can delete the `timeUnit` property.

File: app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/withSavedData.ui.test.js
Comment (bjss-chris) [inline app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/withSavedData.ui.test.js:91]: Typo in on-demand.

File: app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/withSavedData.ui.test.js
Comment (bjss-chris) [inline app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/withSavedData.ui.test.js:149]: We probably ought to rename this data test ID too as `practice` only applies to per patient prices, although it is probably worth checking the impact first – we might want to treat this as tech debt.

File: app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/withoutSavedData.ui.test.js
Comment (bjss-chris) [inline app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/withoutSavedData.ui.test.js:25]: Please see my previous comments on `itemUnit` and `timeUnit`.

**Review by bjss-chris — CHANGES_REQUESTED** [review]
While I was trying to navigate to the edit form I came across an existing bug that prevented me from reaching it, so I've pushed a fix and some tests to your branch.

Once you've made the changes I've requested I'll ask @bjss-andrew to review the PR too as I've also contributed to it now.

All the requested changes are fairly small things but on the whole your PR looks fairly sound.

File: app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/manifest.json
Comment (kiran-mastek) [reply app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/manifest.json]: Done

File: app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/general.ui.test.js
Comment (kiran-mastek) [reply app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/general.ui.test.js]: Done

File: app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/general.ui.test.js
Comment (kiran-mastek) [reply app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/general.ui.test.js]: Done

File: app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/general.ui.test.js
Comment (kiran-mastek) [reply app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/general.ui.test.js]: Done

File: app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/general.ui.test.js
Comment (kiran-mastek) [reply app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/general.ui.test.js]: Done

File: app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/general.ui.test.js
Comment (kiran-mastek) [reply app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/general.ui.test.js]: Done

File: app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/withSavedData.ui.test.js
Comment (kiran-mastek) [reply app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/withSavedData.ui.test.js]: Done

File: app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/withSavedData.ui.test.js
Comment (kiran-mastek) [reply app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/withSavedData.ui.test.js]: Corrected

File: app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/withoutSavedData.ui.test.js
Comment (kiran-mastek) [reply app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/withoutSavedData.ui.test.js]: Done

File: app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/withSavedData.ui.test.js
Comment (kiran-mastek) [reply app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/withSavedData.ui.test.js]: Done

**Review by bjss-chris — APPROVED** [review]
@bjss-andrew this looks good to me but can you also review as I've contributed.

@kiran-mastek I've approved the PR but let's wait for @bjss-andrew to review before merging.

**Review by bjss-andrew — APPROVED** [review]

**Comment by kiran-mastek** [comment]
BULK ORDERING - Edit Catalogue Solution - Flat Variable (On demand) 
Use template to display edit form
Add support and tests for on demand prices

Story AB#9131
Task AB#11902

**Pull request merged**
